src/Workspaces/Core/Portable/FlowAnalysis/SymbolUsageAnalysis/SymbolUsageAnalysis.Walker.cs

```diff
@@ -145,50 +145,61 @@ private void OnReferenceFound(ISymbol symbol, IOperation operation)
                 {
                     OnReadReferenceFound(symbol);
                 }
+            }
 
-                return;
+            private bool MakePendingWrite(IOperation operation, ISymbol symbolOpt)
+            {
+                Debug.Assert(symbolOpt != null || operation.Kind == OperationKind.FlowCaptureReference);
 
-                // Local functions
-                bool MakePendingWrite()
+                if (operation.Parent is IAssignmentOperation assignmentOperation &&
+                    assignmentOperation.Target == operation)
                 {
-                    Debug.Assert(isWrittenTo);
-
-                    if (operation.Parent is IAssignmentOperation assignmentOperation &&
-                        assignmentOperation.Target == operation)
-                    {
-                        var set = PooledHashSet<(ISymbol, IOperation)>.GetInstance();
-                        set.Add((symbol, operation));
-                        _pendingWritesMap.Add(assignmentOperation, set);
-                        return true;
-                    }
-                    else if (operation.IsInLeftOfDeconstructionAssignment(out var deconstructionAssignment))
+                    var set = PooledHashSet<(ISymbol, IOperation)>.GetInstance();
+                    set.Add((symbolOpt, operation));
+                    _pendingWritesMap.Add(assignmentOperation, set);
+                    return true;
```

Comment:
who put this pooled set back?

Reply:
found it.

Reply:
Ah, it should be done here: http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/FlowAnalysis/SymbolUsageAnalysis/SymbolUsageAnalysis.Walker.cs,195, but seems I forgot to release. I will fix it.

Reply:
Actually seems I did it here: http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/FlowAnalysis/SymbolUsageAnalysis/SymbolUsageAnalysis.Walker.cs,73 :)